### PR TITLE
feat: add etcd retry when init

### PR DESCRIPTION
**Pull request by yxudong**
### What this PR does / why we need it:

Fix https://github.com/apache/apisix/issues/3953

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [x] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

**Review by starsz — APPROVED** [review]

File: apisix/cli/etcd.lua [hunk 1]
```diff
@@ -191,10 +191,21 @@ function _M.init(env, args)
...
+            res, err = request(version_url, yaml_conf)
+            -- In case of failure, request returns nil followed by an error message.
+            -- Else the first return value is the response body
+            -- and followed by the response status code.
+            if res then
+                break
+            end
+            retry_time = retry_time + 1
+            print(str_format("Warning! Request etcd endpoint \'%s\' error, %s, retry_time=%s",
+                             version_url, err, retry_time))
```
Comment (spacewander) [inline apisix/cli/etcd.lua:206]: Need a test in https://github.com/apache/apisix/blob/master/t/cli/test_etcd.sh

Comment (spacewander) [reply apisix/cli/etcd.lua:206]: Ping @yxudong

Comment (yxudong) [reply apisix/cli/etcd.lua:206]: > Ping @yxudong

Get it, I‘ll do it soon.

File: t/cli/test_etcd.sh [hunk 2]
```diff
@@ -80,6 +80,24 @@ fi
 
 echo "passed: properly handle the error when connecting to etcd without auth"
 
+# Check etcd retry if connect failed
+git checkout conf/config.yaml
+
+echo '
+etcd:
+  host:
```
Comment (spacewander) [inline t/cli/test_etcd.sh:88]: Better to expand the tab to space

Comment (yxudong) [reply t/cli/test_etcd.sh:88]: > Better to expand the tab to space

Sorry, I‘m not understand.
I just copy other code in same file and do some modify. 
The code style is same. And it's seems no tab in this file.

**Review by spacewander — APPROVED** [review]

**Pull request merged**